src/Symfony/Bundle/FrameworkBundle/Command/SecretsSetCommand.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Bundle\FrameworkBundle\Secrets\SodiumVault;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\ConsoleOutputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * @author Tobias Schultze <http://tobion.de>
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+final class SecretsSetCommand extends Command
+{
+    protected static $defaultName = 'secrets:set';
+
+    private $vault;
+    private $localVault;
+
+    public function __construct(SodiumVault $vault, SodiumVault $localVault = null)
+    {
+        $this->vault = $vault;
+        $this->localVault = $localVault;
+
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setDescription('Sets a secret in the vault.')
+            ->addArgument('name', InputArgument::REQUIRED, 'The name of the secret')
+            ->addArgument('file', InputArgument::OPTIONAL, 'A file where to read the secret from or "-" for reading from STDIN')
+            ->addOption('local', 'l', InputOption::VALUE_NONE, 'Updates the local vault.')
+            ->addOption('random', 'r', InputOption::VALUE_OPTIONAL, 'Generates a random value.', false)
+            ->setHelp(<<<'EOF'
+The <info>%command.name%</info> command stores a secret.
+
+    %command.full_name% <name>
+EOF
+            )
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $errOutput = $output instanceof ConsoleOutputInterface ? $output->getErrorOutput() : $output;
+        $io = new SymfonyStyle($input, $errOutput);
+        $name = $input->getArgument('name');
+        $vault = $input->getOption('local') ? $this->localVault : $this->vault;
+
+        if (null === $vault) {
+            $io->success('The local vault is disabled.');
+
+            return 1;
+        }
+
+        if (0 < $random = $input->getOption('random') ?? 16) {
+            $value = strtr(substr(base64_encode(random_bytes($random)), 0, $random), '+/', '-_');
+        } elseif (!$file = $input->getArgument('file')) {
+            $value = $io->askHidden('Please type the secret value');
+        } elseif ('-' === $file) {
+            $value = file_get_contents('php://stdin');
```

Comment:
Not sure how we can address this, but using pipe will add an extra `LF` at the end of the pipe
`echo 'test' | ./bin/console secrets:set a -` while store `test\n`

Reply:
solution: use `echo -n 'test' | ./bin/console secrets:set a -`

Reply:
that's not the pipe adding the LF, that's `echo`.